### PR TITLE
[AMORO-3117]: Integration of non-Iceberg related format in a pluggable way

### LINKED ISSUE #3117: [Subtask]: Integration of non-Iceberg related format in a pluggable way
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Parent issue

#3116

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

Close #3117.

## Brief change log
 
- Add formats-integration modules
- Move paimon/hudi related code to format-integration/amoro-paimon-integration and format-integration/amoro-hudi-format-integration

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: amoro-core/src/main/java/org/apache/amoro/process/OptimizingStage.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
  * limitations under the License.
  */
 
-package org.apache.amoro.api.process;
+package org.apache.amoro.process;
```
Comment (zhoujinsong) [inline amoro-core/src/main/java/org/apache/amoro/process/OptimizingStage.java:19]: What is the reason for moving these 3 classes from module `amoro-ams-api` to module `amoro-core`?

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/dashboard/MixedAndIcebergTableDescriptor.java [hunk 2]
```diff
@@ -92,10 +101,11 @@ public class MixedAndIcebergTableDescriptor extends PersistentBase
 
   private static final Logger LOG = LoggerFactory.getLogger(MixedAndIcebergTableDescriptor.class);
 
-  private final ExecutorService executorService;
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/dashboard/MixedAndIcebergTableDescriptor.java:95]: Maybe we should move this implementation to module `amoro-iceberg-format`?
If yes, we can do it in the next PR.

**Review by zhoujinsong — COMMENTED** [review]
I left some comments, PTLA. @baiyangtx

**Comment by zhoujinsong** [comment]
AFAIK, after this change AMS does not support the Paimon/Hudi table format by default.
Users may need to add the Paimon/Hudi dependencies manually, am I right? @baiyangtx

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/dashboard/MixedAndIcebergTableDescriptor.java (hunk 2 above)
Comment (baiyangtx) [reply amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/dashboard/MixedAndIcebergTableDescriptor.java:95]: yes

File: amoro-core/src/main/java/org/apache/amoro/process/OptimizingStage.java (hunk 1 above)
Comment (baiyangtx) [reply amoro-core/src/main/java/org/apache/amoro/process/OptimizingStage.java:19]: TableDescriptor depends on those class, and the TableDescriptor was moved to amoro-core

**Comment by baiyangtx** [comment]
> AFAIK, after this change AMS does not support the Paimon/Hudi table format by default. Users may need to add the Paimon/Hudi dependencies manually, am I right? @baiyangtx

NO, I forget the profile parts. I will add that.

File: amoro-core/src/main/java/org/apache/amoro/process/OptimizingStage.java (hunk 1 above)
Comment (zhoujinsong) [reply amoro-core/src/main/java/org/apache/amoro/process/OptimizingStage.java:19]: Shall we move other classes into module `amoro-core` too in the next PR?
If yes, this make sense to me.

File: amoro-ams/amoro-ams-server/pom.xml [hunk 3]
```diff
@@ -472,4 +472,38 @@
             </plugin>
         </plugins>
     </build>
+
+    <profiles>
+        <profile>
+            <id>paimon</id>
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/pom.xml]: `support-paimon-format`,`support-hudi-format`,`support-all-formats` may be proper names.

**Review by zhoujinsong — APPROVED** [review]
LGTM. Thanks for the contribution.

**Review by czy006 — APPROVED** [review]

**Pull request merged**